### DISCUSSION TITLE
Mod Loader should follow the same rules for Modded badge as standard one

### DISCUSSION THREAD

**Discussion by StreetStrider** [trumank/mint#208] [General]
I'm using a bunch of client-only QoL mods regularily and I host games. All my mods are Verified + Optional and no other tags. I play non-gameplay-changing environment and I want that people with the same mindset to actively join my games. However, they see Modded badge on my server description. I think a big fracture of players ignore my server now, since they do not want modded game or afraid of mods' side-effects, like affecting game save. That is a big problem for me.

I think Mod Loader should follow the same rules as the vanilla one. I've never seen blue Modded tag when using mod.io integration in-game. With Mod Loader and the same list of mods my server implicitly marked as Modded.

I think the proper criteria would be to filter out all Verified + Optional mods in list and if the result is not empty then add the badge, otherwise no badge.

**Comment by Trava715** [comment]
If anything it should be a switch in the mod loader. I'm using a bunch of approved mods that do not affect other people's gameplay in any meaningful way (eg. chat popups for pressing buttons, Inferno heat bug fix, grenade collision fix, removing mineral scatter from TCF)  or clientside approved mods (better flashlight, more flares) and I do not think that it warrants the MODDED tag since the gameplay is very much still vanilla and as we all know, people tend to avoid modded lobbies due to the awful official integration that makes you subscribe to hosts' mods and leave them enabled upon leaving the lobby.

**Comment by jieyouxu** [comment]
The likely path mint is going to take is likely to be a bit more conservative than suggested:
- We (most likely) will try to implement *auto-verification*, in the same vein as how mod.io does auto-verification. This is a conservative approach, because while some mods may become *verified* under *manual approval*, mint or mod.io cannot automatically determine so.
- But mint, unlike mod.io, has to deal with `.pak`s sourced from the internet or locally from disk, which means that we want to treat these `.pak`s in the same way as mod.io paks.

It *might* be a possibility for us to consider also fully respecting manual *verified* labels from mod.io, even though the same pak uploaded to mod.io when used locally might not be able to be automatically verified.

An edge case might be a *approved* mod from mod.io for some balance reason that is manually marked as approved but actually can be auto-verified. In this case, mint will likely determine the mod as auto-verified still.

**EDIT**: we will respect mod.io manual verification as well.

(This is *not* the final decision, and is still subject to further changes.)

**Comment by jieyouxu** [comment]
Auto-verification is partially implemented in PR #124, but more testing is required and integration logic is not fully complete.

If a profile consists of *only* verified (either auto-verified or manually verified from mod.io) then this is a *verified profile*, so no `[MODDED]` tag will be added. Note that in this case, some mods when used locally as `.pak`s may not be able to be auto-verified but could be manually verified on mod.io. We respect the mod.io manual verified labels.

**Comment by StreetStrider** [comment]
I've recently found another case where it is a problem. Some people start joining my lobbies expecting some heavy gameplay modding. After explaining this is pure server (labeled by mistake) they were discouraged and sometimes leave immediately. So this issue is officially double-edged now.

**Comment by StreetStrider** [comment]
Please, consider this idea.

**Comment by FenekkuKitsune** [comment]
I personally think the switch idea has merit. I use the Server Name mod which automatically changes my server's name based on in-game criteria such as mission type, warnings, etc. I *think* it overrides MINT's server name modded tagging (I can't tell, it's not my intention anyway). I manually add the [MODDED] tag to all my titles in the Server Name mod's filters.

I don't mind who joins and who doesn't join, personally, I've not noticed too major a difference in who does and doesn't join (though some people don't read the server title and ask "Oh is this modded?" after seeing multiple MULEs). But, I do agree that it would be nice to have no tag when the mods are clientside.

Adding logic to detect for verified vs approved/sandbox is a good step, but relying on mod.io's verification process is a double-edged sword. On the one hand, you have a very easy and reliable way to match DRG's official mod integration. On the other hand, mod.io's verification process has been severely lacking lately, with a lot of mods waiting weeks for verification, outright being ignored entirely, or being mislabeled. There's a reason a lot of us have turned to MINT, after all.
